mono/mini/debugger-agent.c

```diff
@@ -6055,12 +6149,12 @@ ss_start_il2cpp(SingleStepReq *ss_req, DebuggerTlsData *tls)
 	// Recreating this on each pass is a little wasteful but at least keeps behavior linear.
 	int ss_req_bp_count = g_slist_length(ss_req->bps);
 	GHashTable *ss_req_bp_cache = NULL;
+	gboolean enable_global = FALSE;
 
 	/* Stop the previous operation */
 	ss_stop(ss_req);
 
 	DEBUG_PRINTF(0, "Step depth: %d\n", ss_req->depth);
-	g_assert(ss_req->depth != STEP_DEPTH_INTO);
```

Comment:
Why can we remove this now?

Reply:
This check was never in the original ss_start code.  There is code below this (brought over from the original mono code) that checks for STEP_DEPTH_INTO and sets the global breakpoint flag.